rotkehlchen/chain/ethereum/modules/l2/zksync.py

```diff
@@ -0,0 +1,589 @@
+                raise RemoteError(msg)
+
+            for idx, entry in enumerate(result):
+                if idx == 0 and entry['txHash'] == last_tx_hash:
+                    continue  # at pagination first tx is last query's last
+
+                tx = self._decode_zksync_transaction(entry)
+                if tx:
+                    transactions.append(tx)
+                    last_tx_hash = entry['txHash']
+
+            yield transactions
+            if len(result) < ZKSYNCLITE_MAX_LIMIT:
+                break  # no need to paginate further
+
+            from_hash = result[-1]['txHash']
+            transactions = []
+
+    def _add_zksynctxs_db(
+            self,
+            write_cursor: 'DBCursor',
+            transactions: Iterable[ZKSyncLiteTransaction],
+    ) -> None:
+        tuples = [x.serialize_for_db() for x in transactions]
+        query = """INSERT INTO zksynclite_transactions(tx_hash, type, timestamp, block_number,
+        from_address, to_address, token_identifier, amount, fee)
+        VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)"""
+        self.database.write_tuples(
+            write_cursor=write_cursor,
+            tuple_type='zksynclite_transaction',
+            query=query,
+            tuples=tuples,
+        )
+
+    def _get_zksynctxs_db(
+            self,
+            address: ChecksumEvmAddress,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[ZKSyncLiteTransaction]:
+        cursor = self.database.conn.cursor()
+        result = cursor.execute(
+            'SELECT * from zksynclite_transactions WHERE timestamp >= ? AND timestamp <= ? '
+            'AND (from_address == ? OR to_address == ?) ORDER by timestamp DESC',
+            (start_ts, end_ts, address, address),
+        )
+        transactions = []
+        for entry in result:
+            try:
+                transactions.append(ZKSyncLiteTransaction.deserialize_from_db(entry))
+            except (DeserializationError, UnknownAsset) as e:
+                log.error(
+                    f'Could not deserialize zksync lite transaction {entry} from the DB due to {e!s}',  # noqa: E501
+                )
+                continue
+
+        return transactions
+
+    def get_transactions(
+            self,
+            address: ChecksumEvmAddress,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[ZKSyncLiteTransaction]:
+        """Get all zksync transactions for an address in the given time range"""
+        location = f'zksynctxs_{address}'
+        min_tx_hash = max_tx_hash = 'latest'
+        min_timestamp = start_ts
+        max_timestamp = end_ts
+
+        with self.database.conn.read_ctx() as cursor:
+            queried_range = self.database.get_used_query_range(cursor, location)
+
+            cursor.execute('SELECT tx_hash, min(timestamp) from zksynclite_transactions;')
+            result = cursor.fetchone()
+
+            if result[0]:
```

Comment:
can't this return `None` and you need a check for `if result is None: something`